src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,85 @@
-export const Pagination = () => {};
+import React from 'react';
+import classNames from 'classnames';
+
+const getPages = (max: number) => {
+  const pagesArray = [];
+
+  for (let i = 1; i <= max; i += 1) {
+    pagesArray.push(i);
+  }
+
+  return pagesArray;
+};
+
+type Props = {
+  total: number,
+  perPage: number,
+  currentPage: number,
+  onPageChange: (page: number) => void,
+};
+
+export const Pagination: React.FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const maxPage = Math.ceil(total / perPage);
+  const pages = getPages(maxPage);
+
+  const handlePageSelect = (page: number) => {
+    if (page !== currentPage && page > 0 && page <= maxPage) {
+      onPageChange(page);
+    }
+  };
+
+  return (
+    <ul className="pagination">
+      <li className={
+        classNames('page-item', { disabled: currentPage === 1 })
+      }
+      >
+        <a
+          data-cy="prevLink"
+          className="page-link"
+          href="#prev"
+          aria-disabled={currentPage === 1}
+          onClick={() => handlePageSelect(currentPage - 1)}
+        >
+          «
+        </a>
+      </li>
+
+      {pages.map(page => (
+        <li className={
+          classNames('page-item', { active: page === currentPage })
+        }
+        >
```

Comment:
don't forget about `key` by mapping
```suggestion
        <li
          key={page}
          className={classNames('page-item', { active: page === currentPage })}
        >
```